File: solutions/ocaml/scrabble-score/1/scrabble_score.ml
let score word =
  let score_letter = function
    | 'D' | 'G' -> 2
    | 'B' | 'C' | 'M' | 'P' -> 3
    | 'F' | 'H' | 'V' | 'W' | 'Y' -> 4
    | 'K' -> 5
    | 'J' | 'X' -> 8
    | 'Q' | 'Z' -> 10
    | _ -> 1
  in
  String.fold_left
    (fun acc c -> acc + score_letter c)
    0
    (String.uppercase_ascii word)

File: solutions/clojure/scrabble-score/1/src/scrabble_score.clj
(ns scrabble-score
    (:require [clojure.string :as cstr]))

(defn score-letter
  [letter]
  (condp #(cstr/includes? %1 %2) (cstr/upper-case letter)
    "AEIOULNRST" 1
    "DG"         2
    "BCMP"       3
    "FHVWY"      4
    "K"          5
    "JX"         8
    "QZ"         10))


(defn score-word
  [word]
  (apply + (map score-letter word)))
